### ISSUE TITLE
Paginated pages using query string are not scrapped when using "bySiteStructure"

### ISSUE THREAD

**Issue by Cerlancism** [website-scraper/node-website-scraper#411]
### Configuration
**version:** -- website-scraper@4.2.1

**options:** 
```js
/**
 * @type {import("website-scraper").Options}
 */
const options = {
    urls: ["http://testing-ground.scraping.pro"],
    directory: 'scrapped',
    recursive: true,
    maxRecursiveDepth: 3,
    filenameGenerator: "bySiteStructure",
};


```

### Description
Websites with paginated pages using query strings are not scrapped properly.
Examples:  
http://testing-ground.scraping.pro/blocks?ver=1
http://testing-ground.scraping.pro/blocks?ver=2
http://testing-ground.scraping.pro/blocks?ver=3
http://testing-ground.scraping.pro/blocks?ver=4
http://testing-ground.scraping.pro/blocks?ver=5

**Expected behavior:** All these pages are scrapped to the site structure path

**Actual behavior:** Pages paginated with query string are always being rewritten to the same `blocks/index.html` of the site structure path

### Additional Information
I have come up a folk with a simple workaround to generate the file names as `blocks/index_ver=1.html` etc

**Comment by s0ph1e** [comment]
Hi @Cerlancism 👋 

Thank you for describing this problem. Looks like `bySiteStructure` filename generator can be improved to save different files for different query params. 

Unfortunately I do not have time to do it now so feel free to contribute

**Comment by s0ph1e** [comment]
Issue was fixed in version 4.2.2
